src/victory-util/helpers.js

```diff
@@ -144,7 +145,18 @@ export default {
       return (x) => x;
     }
     // otherwise, assume it is an array index, property key or path (_.property handles all three)
-    return property(key);
+    return (x) => {
+      if (Immutable.isImmutable(x)) {
+        if (Array.isArray(key)) {
+          return x.getIn(key);
+        } else if (typeof key === "string") {
+          return x.getIn(key.replace("[", ".").replace("]", "").split("."));
+        } else {
+          return x.get(key);
+        }
+      }
+      return property(key)(x);
+    };
```

Comment:
This is more of a nice-to-have. People could just always specify an accessor function if they are passing in an `immutable` object.